### PR TITLE
feat: cherry-pick author attribution and PR assignment

### PR DESCRIPTION
**Pull request by rnetser**
## Summary
- Add PR author attribution for cherry-pick PRs (issue #993)
- Assign cherry-picked PRs to the original PR author instead of the requester
- Add schema validation for `cherry-pick-assign-to-pr-author` config option

## Test plan
- [ ] Cherry-pick attribution tests pass
- [ ] Schema validation tests pass

### CODE REVIEW

File: webhook_server/tests/test_runner_handler.py [hunk 1]
```diff
@@ -918,6 +919,164 @@ async def test_cherry_pick_manual_needed(self, runner_handler, mock_pull_request
...
+                                    )
+                                    mock_set_progress.assert_called_once()
+                                    mock_set_success.assert_called_once()
+                                    mock_comment.assert_called_once()
+                                    # Verify the hub command's last -m contains the by_label format
+                                    last_cmd = mock_run_cmd.call_args_list[-1]
+                                    hub_command = last_cmd.kwargs.get(
+                                        "command", last_cmd.args[0] if last_cmd.args else ""
+                                    )
+                                    assert "requested-by by label-user with target-branch label" in hub_command
```
[the remark this answers is not in the document]
Comment (rnetser) [reply webhook_server/tests/test_runner_handler.py]: Done - Added `test_cherry_pick_by_label_empty_reviewed_user_requested_by_format` to cover the edge case. Asserts clean `requested-by by target-branch label` and no double-space bug.

**Comment by rnetser** [comment]
> `@rnetser`: I've reviewed PR `#994`. Everything looks excellent:
> 
> **✅ Implementation Quality**
> 
> * **Clean architecture**: Config option properly defined in schema with global/per-repo hierarchy
> * **Simplified API**: `cherry_pick()` signature improved—`assign_to_pr_owner` is more explicit than `reviewed_user`
> * **Proper attribution**: Cherry-pick PR descriptions now include source branch and original PR owner
> * **Sensible default**: `true` ensures proper attribution out-of-the-box
> 
> **✅ Testing & Quality**
> 
> * **Comprehensive coverage**: All handlers, config loading, and validators tested
> * **1292 tests passing** with 90.50% coverage
> * **Ruff checks clean**—formatting and linting standards met
> * **New test cases**: `test_cherry_pick_assigns_pr_author` and `test_cherry_pick_requested_by_uses_pr_owner` validate the feature
> 
> **✅ Configuration & Documentation**
> 
> * Schema validation updated for the new boolean field
> * Examples updated in both config formats
> * Follows existing configuration patterns
> 
> **No issues found.** The feature implementation is solid, well-tested, and maintains consistency with the codebase patterns.
> 
> 🧠 Learnings used
> ```
> Learnt from: myakove
> Repo: myk-org/github-webhook-server PR: 612
> File: webhook_server_container/libs/github_api.py:2089-2100
> Timestamp: 2024-10-29T08:09:57.157Z
> Learning: In `webhook_server_container/libs/github_api.py`, when the function `_keep_approved_by_approvers_after_rebase` is called, existing approval labels have already been cleared after pushing new changes, so there's no need to check for existing approvals within this function.
> ```
> 
> ```
> Learnt from: myakove
> Repo: myk-org/github-webhook-server PR: 1002
> File: webhook_server/tests/test_pull_request_review_handler.py:298-344
> Timestamp: 2026-02-28T12:05:25.906Z
> Learning: In the myk-org/github-webhook-server repository's test-oracle integration (webhook_server/libs/test_oracle.py and webhook_server/libs/handlers/*.py), the 'approved' trigger refers to the `/approve` **command** (user-initiated comment command), NOT GitHub's review approval state. GitHub review approvals without the `/approve` command should NOT trigger the test oracle. This distinction is critical: review state "approved" alone does not trigger oracle; only the explicit `/approve` command does.
> ```
> 
> ✅ Actions performed

@coderabbitai

if all looks ood, approve the pr
